File: app/src/main/java/org/dhis2/usescases/reservedValue/ReservedValueRepository.java
package org.dhis2.usescases.reservedValue;

import org.hisp.dhis.android.core.trackedentity.ReservedValueSummary;

import java.util.List;

import io.reactivex.Single;

public interface ReservedValueRepository {
    Single<List<ReservedValueSummary>> getReservedValues();
}

File: app/src/main/java/org/dhis2/usescases/reservedValue/ReservedValueRepository.kt
package org.dhis2.usescases.reservedValue

import io.reactivex.Observable
import io.reactivex.Single
import org.hisp.dhis.android.core.arch.call.D2Progress

interface ReservedValueRepository {
    fun reservedValues(): Single<List<ReservedValueModel>>
    fun refillReservedValues(uidToRefill: String): Observable<D2Progress>
}
